### PR TITLE
Updated mem backend to comply with new io integration test suite at b…

### PR DESCRIPTION
**Pull request by funkyshu**
…ackend/testsuiite/io_integration_test.go

### CODE REVIEW

File: backend/mem/file.go [hunk 1]
```diff
@@ -14,223 +13,217 @@ import (
...
-		if err != nil {
-			return 0, err
-		}
-		err = f.Touch()
-		if err != nil {
-			return 0, err
+	if f.writeMode == none {
+		// unless seek or read is called first, writes should replace a file (not edit)
+		if f.readCalled || f.seekCalled {
+			f.readWriteSeeker = NewReadWriteSeekerWithData(f.memFile.contents)
```
Comment (funkyshu) [inline backend/mem/file.go:182]: In this case, we ope in edit mode

File: backend/mem/readwriteseeker.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	if position < 0 || position > int64(len(rws.data)) {
+		return 0, vfs.ErrSeekInvalidOffset
+	}
+
+	rws.cursor = int(position)
+
+	return position, nil
+}
+
+func (rws *ReadWriteSeeker) Bytes() []byte {
```
Comment (dustin-schnee) [inline backend/mem/readwriteseeker.go:78]: Missing comment here.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 ### Added
 - Fixes #156 - Update os backend to comply with new io integration test suite
+- Fixes #160 - Update os backend to comply with new io integration test suite
```
Comment (dustin-schnee) [inline CHANGELOG.md:9]: nit: should be `Update mem backend to comply with new io integration test suite`

**Review by dustin-schnee — CHANGES_REQUESTED** [review]
Just a couple nit picks.

**Review by dustin-schnee — APPROVED** [review]

**Pull request merged**